js/protocols/stm32.js

```diff
@@ -342,6 +344,8 @@ STM32_protocol.prototype.verify_flash = function (first_array, second_array) {
 STM32_protocol.prototype.upload_procedure = function (step) {
     var self = this;
 
+    var extendedErase=false;
```

Comment:
this and the other references to 'extendedErase' needed to be `self.extendedErase`.